telosnetwork/telos-wallet
# EVM Metakeep prototype

# Fixes #718

This PR includes the implementation of the potential replacement for Telos Cloud using MetaKeep as the main signing service.

**EVM:**  A new EVMAuthenticator was created that fits perfectly into the class hierarchy we have for EVM.
**ZERO:** A new UAL Authenticator was created for the Native chain wrapping Metakkep service.

In both cases, the login process will take much longer the first time than subsequent times. This is because the account data is consulted the first time and then cached to avoid future queries (since it is data that will never change).
In the particular case of Telos Zero, we also have the creation of the account that is done only once, the first time the user authenticates.


Test Scenarios:
- Telos EVM
  - go to [this link](https://deploy-preview-709--wallet-staging.netlify.app/)
  - Press Telos Cloud Wallet button
  - wait for the google button to render and click on it
    - A window should popup and after you identify yourself it will close and put the button in loading mode until you are redirected to balance page
  - Once logged, try to perform any transaction.
    - The first time, you will be prompted to enter a 6-digit number that will be sent to your email
    - After you confirm that, the following transactions will skip this step
- If you refresh the page (press F5) you should be auto-logged as expected.

- Telos Zero
  - go to [this link](https://deploy-preview-709--wallet-staging.netlify.app/?login=zero)
  - Press Telos Cloud Wallet button
  - wait for the google button to render and click on it
    - A window should pop up and after you identify yourself it will close and put the button in loading mode until you are redirected to balance page.
    - remember first time doing this will take much longer than the following opportunities.
  - Once logged, try to perform any transaction.
    - The first time, you will be prompted to enter a 6-digit number that will be sent to your email
    - After you confirm that, the following transactions will skip this step
- If you refresh the page (press F5) you should be auto-logged as expected.
